addons/lunch/models/lunch_alert.py

```diff
@@ -88,30 +88,32 @@ def _search_available_today(self, operator, value):
         ])
 
 
-    @api.depends('name', 'mode', 'notification_time', 'notification_moment', 'tz', 'active')
-    def _compute_cron_id(self):
-        for alarm in self:
-            alarm = alarm.with_context(tz=alarm.tz)
+    def _sync_cron(self):
+        """ Synchronise the related cron fields to reflect this alert """
+        for alert in self:
+            alert = alert.with_context(tz=alert.tz)
 
             cron_required = (
-                alarm.active
-                and alarm.mode == 'chat'
-                and (not alarm.until or fields.Date.context_today(alarm) <= alarm.until)
+                alert.active
+                and alert.mode == 'chat'
+                and (not alert.until or fields.Date.context_today(alert) <= alert.until)
             )
 
-            now_tz = fields.Datetime.context_timestamp(alarm, datetime.utcnow())
-            sendat_tz = datetime.combine(date.today(), float_to_time(alarm.notification_time, alarm.notification_moment, alarm.tz))
-            if sendat_tz < now_tz:  # notif sent today already
+            sendat_tz = datetime.combine(
+                fields.Date.context_today(alert, fields.Datetime.now()),
+                float_to_time(alert.notification_time, alert.notification_moment, alert.tz))
```

Comment:
I use `fields.Date.context_today(alert, fields.Datetime.now())` instead of `fields.Date.context_today(alert)` because I can mock `fields.Datetime.now` in the tests.